ompi/mca/pml/ucx/pml_ucx.c

```diff
@@ -193,14 +193,23 @@ int mca_pml_ucx_init(void)
     ucp_worker_params_t params;
     ucp_worker_attr_t attr;
     ucs_status_t status;
+    ucp_context_attr_t ctx_attr;
     int i, rc;
 
     PML_UCX_VERBOSE(1, "mca_pml_ucx_init");
 
+    /* request for thread mode supported by UCX */
+    attr.field_mask = UCP_ATTR_FIELD_THREAD_MODE;
+    status = ucp_context_query(ompi_pml_ucx.ucp_context, &ctx_attr);
+    if (status != UCS_OK) {
+        PML_UCX_ERROR("Failed to query UCP context status");
+        rc = OMPI_ERROR;
+        goto err;
+    }
+
     /* TODO check MPI thread mode */
-    params.field_mask  = UCP_WORKER_PARAM_FIELD_THREAD_MODE;
-    params.thread_mode = UCS_THREAD_MODE_SINGLE;
-    if (ompi_mpi_thread_multiple) {
+    params.field_mask = UCP_WORKER_PARAM_FIELD_THREAD_MODE;
+    if (ompi_mpi_thread_multiple && (ctx_attr.thread_mode == UCS_THREAD_MODE_MULTI)) {
```

Comment:
didn't get... in case if multi-thread is requested - then check for MT is supported by UCX, and make decision about worker attributes. Actually MPI allows to fallback into single thread mode even in case if Multi-Thread is requested (actual mode is returned)

Reply:
`ctx_attr.thread_mode == UCS_THREAD_MODE_MULTI` does not indicate if MT is supported by UCX

Reply:
hmmmmm... not good. will fix